From 653e60e93a016b4b12849e0ab19954ecefb26865 Mon Sep 17 00:00:00 2001
From: Dinesh Kumar Mirche <dmirche@habana.ai>
Date: Thu, 25 Jul 2024 05:32:42 +0300
Subject: [PATCH] [SW-195702] gaudi3: sec: open jtag interface

This patch opens the JTAG interface when security is
enabled and JTAG_DIS efuse bit is zero.

The exports.elf file is generated from arcboot_rom.elf
which is generated from released arcboot ROM i.e
git tag "gaudi3_secureboot_rom_tapeout_final_drop_3"

Change-Id: I60227ad5d88bf1de403bf7c1a469bebb0b85b04e
Signed-off-by: Dinesh Kumar Mirche <dmirche@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 8317717..ed7d4ce 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1311,8 +1311,14 @@
 
 void hl_dev_security_init(void)
 {
-	if (ppboot_sys_cfg_efuse_is_set(EFUSE_SEC_EN_MASK))
+	if (ppboot_sys_cfg_efuse_is_set(EFUSE_SEC_EN_MASK)) {
+		/* When security is enabled and JTAG_DIS bit is
+		 * not set in fuse, JTAG IF is opened for debugging.
+		 */
+		security_handle_jtag_if();
+		security_rr_open_debug_if_config();
 		return;
+	}
 
 	security_rr_open_pcie_config();
 }
diff --git a/arch/gaudi3/arc/exports.elf b/arch/gaudi3/arc/exports.elf
index af5d207..5be0aee 100644
--- a/arch/gaudi3/arc/exports.elf
+++ b/arch/gaudi3/arc/exports.elf
Binary files differ
